### DISCUSSION TITLE
questions about DeepMD permutation symmetry & network

### DISCUSSION THREAD

**Discussion by chandagsin** [deepmodeling/deepmd-kit#2352] [Q&A]
Dear Developers, 

one of the key points of DeepMD is to construct a descriptor that preserves the permutation symmetry. I am new to neural network. It appears to me that you have proposed two different descriptors for the permutation symmetry (and I am a bit confused about  this). 

(1)   {D_ij} = {1/r_ij, x_ij/r_ij, y_ij/r_ij, z_ij/r_ij},  (e.g, Wang et al, DeepMD-kit, Comp. Physics Comm. 228 (2018) 178–184)
(2)   a network that maps the original local coordinate to the descriptor. the descriptor satisfies the permutation symmetry by its own structure. (e.g., Zhang et al., 2018 DeepPot-SE )

From what I understand, (1) itself is not permutation symmetric unless enough samples are taken from the configuration space, since molecules are not perfectly rigid, e.g., a CH4 molecule may not be a regular tetrahedron (may be distorted and hence the H- atoms symmetry is broken) at a specific point of time. 

**I would like to know, if the current DeepMD-kit actually adopts (2).** the example JSON files specify the descriptor as (e.g.), 
	"descriptor" :{
	    "type":		"se_e2_a",
	    "sel":		[46, 92],
	    "rcut_smth":	5.80,
	    "rcut":		6.00,
	    "neuron":		[25, 50, 100],
	    "resnet_dt":	false,
	    "axis_neuron":	16,
	    "seed":		1,
	    "_comment":		" that's all"
	},

I think this "neuron" would be the network for the construction of descriptor, as detailed in Zhang et al., 2018. 

**I also want to know what would be proper sizes ("neuron") for the embedding net and the fitting net?**

**I guess the "axis_neuron" has something to do with the "G" matrix in Zhang's paper. what would be a good value for it?**

Thanks!

**Comment by njzjz** [comment]
Both 1 (`loc_frame`) and 2 (`se_e2_a`) are available, and the descriptor is decided by setting `descriptor/type`. For more information, please read [documentation](https://docs.deepmodeling.com/projects/deepmd/en/v2.2.0.b0/model/overall.html).

For `neuron` please read the [FAQ](https://docs.deepmodeling.com/projects/deepmd/en/v2.2.0.b0/troubleshooting/howtoset_netsize.html).

**Reply by chandagsin** [reply]
![20230301_2](https://user-images.githubusercontent.com/122434202/222135942-9f8c8628-8c32-4959-b1cc-c91ca6037dcb.PNG)
Thanks!. for Axis_neuron, I guess it's the number of columns of matrix G. what would be a good value?

**Reply by njzjz** [reply]
We've never benchmarked `axis_neuron`. You can start with the value that you posted.
